sql/drop.go

```diff
@@ -87,3 +88,67 @@ func (p *planner) DropTable(n *parser.DropTable) (planNode, error) {
 	}
 	return &valuesNode{}, nil
 }
+
+// DropDatabase drops a database.
```

Comment:
Our `DROP DATABASE` is like the postgres `DROP SCHEMA` (cockroach database == postgres schema). I think the postgres default of not dropping the schema if there are dependent objects is more sensible (see the `RESTRICT` and `CASCADE` options). This doesn't need to be addressed in this PR other than a TODO.